			     BASH PATCH REPORT
			     =================

Bash-Release:	4.0
Patch-ID:	bash40-017

Bug-Reported-by:	Lubomir Rintel <lkundrak@v3.sk>
Bug-Reference-ID:	<1237654931.32737.13.camel@localhost.localdomain>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2009-03/msg00174.html

Bug-Description:

Adding a null line to a here-document (e.g., by hitting EOF) causes the
shell to dump core attempting to dereference the NULL pointer.

Patch:

*** bash-4.0/parse.y	2009-03-08 21:24:47.000000000 -0400
--- bash-4.0-patched/parse.y	2009-03-21 14:38:42.000000000 -0400
***************
*** 1880,1884 ****
    ret = read_a_line (remove_quoted_newline);
  #if defined (HISTORY)
!   if (remember_on_history && (parser_state & PST_HEREDOC))
      {
        /* To make adding the the here-document body right, we need to rely
--- 1880,1884 ----
    ret = read_a_line (remove_quoted_newline);
  #if defined (HISTORY)
!   if (ret && remember_on_history && (parser_state & PST_HEREDOC))
      {
        /* To make adding the the here-document body right, we need to rely
*** bash-4.0/patchlevel.h	2009-01-04 14:32:40.000000000 -0500
--- bash-4.0-patched/patchlevel.h	2009-02-22 16:11:31.000000000 -0500
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 16
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 17
  
  #endif /* _PATCHLEVEL_H_ */
